components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIMappingUtil.java

```diff
@@ -1882,6 +1903,12 @@ public static AsyncAPISpecificationValidationResponseDTO getAsyncAPISpecificatio
                 infoDTO.setDescription(modelInfo.getDescription());
                 infoDTO.setEndpoints(modelInfo.getEndpoints());
                 infoDTO.setProtocol(model.getProtocol());
+                if (SolaceConstants.SOLACE_ENVIRONMENT.equals(new SolaceApiParser().getVendorFromExtension(model.
```

Comment:
do we need this solace related implementation?

Reply:
Yes. To show the available endpoint URLs on the publisher portal for Solace deployments we need to use the Solace Admin APIs. Hence this part is needed.